### ISSUE TITLE
xread with block will send at the same channel  when use AsyncConnectionPoolSupport

### ISSUE THREAD

**Issue by xingzhang8023** [lettuce-io/lettuce-core#1913]
## Bug Report

   @Autowired
   private AsyncPool<ProxyConnection<String, String>> connectionAsyncPool  // generate by "AsyncConnectionPoolSupport.createBoundedObjectPool"

It is expected that a new connection can be obtained each time. However, during the actual test, multiple xreads are concurrently in the same channel.
#### Current Behavior
Multiple xreads with the block parameter are sent in the same channel. As a result, subsequent commands can be executed only after the previous commands are responded.

```
2021/11/11 10:53:18.723 [] [lettuce-epollEventLoop-8-5] [DEBUG] i.l.c.p.DefaultEndpoint - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800, epid=0x4b] write() writeAndFlush command ClusterCommand [command=SubscriptionCommand [type=XREAD, output=StreamReadOutput [output=[], error='null'], commandType=io.lettuce.core.protocol.TracedCommand], redirections=0, maxRedirections=5]
2021/11/11 10:53:18.723 [] [lettuce-epollEventLoop-8-5] [DEBUG] i.l.c.p.DefaultEndpoint - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800, epid=0x4b] write() done
2021/11/11 10:53:18.723 [] [lettuce-epollEventLoop-8-3] [DEBUG] i.l.c.p.CommandHandler - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800, epid=0x4b, chid=0x4b] write(ctx, ClusterCommand [command=SubscriptionCommand [type=XREAD, output=StreamReadOutput [output=[], error='null'], commandType=io.lettuce.core.protocol.TracedCommand], redirections=0, maxRedirections=5], promise)
2021/11/11 10:53:18.723 [] [lettuce-epollEventLoop-8-3] [DEBUG] i.l.c.p.CommandEncoder - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800] writing command ClusterCommand [command=SubscriptionCommand [type=XREAD, output=StreamReadOutput [output=[], error='null'], commandType=io.lettuce.core.protocol.TracedCommand], redirections=0, maxRedirections=5]
2021/11/11 10:53:18.723 [] [lettuce-epollEventLoop-8-3] [TRACE] i.l.c.p.CommandEncoder - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800] Sent: *8^M


2021/11/11 10:53:26.950 [] [lettuce-epollEventLoop-8-5] [DEBUG] i.l.c.p.DefaultEndpoint - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800, epid=0x4b] write() writeAndFlush command ClusterCommand [command=SubscriptionCommand [type=XREAD, output=StreamReadOutput [output=[], error='null'], commandType=io.lettuce.core.protocol.TracedCommand], redirections=0, maxRedirections=5]
2021/11/11 10:53:26.950 [] [lettuce-epollEventLoop-8-5] [DEBUG] i.l.c.p.DefaultEndpoint - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800, epid=0x4b] write() done
2021/11/11 10:53:26.950 [] [lettuce-epollEventLoop-8-3] [DEBUG] i.l.c.p.CommandHandler - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800, epid=0x4b, chid=0x4b] write(ctx, ClusterCommand [command=SubscriptionCommand [type=XREAD, output=StreamReadOutput [output=[], error='null'], commandType=io.lettuce.core.protocol.TracedCommand], redirections=0, maxRedirections=5], promise)
2021/11/11 10:53:26.950 [] [lettuce-epollEventLoop-8-3] [DEBUG] i.l.c.p.CommandEncoder - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800] writing command ClusterCommand [command=SubscriptionCommand [type=XREAD, output=StreamReadOutput [output=[], error='null'], commandType=io.lettuce.core.protocol.TracedCommand], redirections=0, maxRedirections=5]
2021/11/11 10:53:26.950 [] [lettuce-epollEventLoop-8-3] [TRACE] i.l.c.p.CommandEncoder - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800] Sent: *8^M



2021/11/11 10:54:18.836 [] [lettuce-epollEventLoop-8-3] [DEBUG] i.l.c.p.CommandHandler - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800, epid=0x4b, chid=0x4b] Received: 5 bytes, 2 commands in the stack
2021/11/11 10:54:18.836 [] [lettuce-epollEventLoop-8-3] [TRACE] i.l.c.p.CommandHandler - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800, epid=0x4b, chid=0x4b] Buffer: *-1
2021/11/11 10:54:18.836 [] [lettuce-epollEventLoop-8-3] [DEBUG] i.l.c.p.CommandHandler - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800, epid=0x4b, chid=0x4b] Stack contains: 2 commands
2021/11/11 10:54:18.836 [] [lettuce-epollEventLoop-8-3] [DEBUG] i.l.c.p.CommandHandler - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800, epid=0x4b, chid=0x4b] Completing command LatencyMeteredCommand [type=XREAD, output=StreamReadOutput [output=[], error='null'], commandType=io.lettuce.core.cluster.ClusterCommand]


2021/11/11 10:54:31.291 [] [lettuce-epollEventLoop-8-3] [DEBUG] i.l.c.p.CommandHandler - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800, epid=0x4b, chid=0x4b] Received: 108 bytes, 1 commands in the stack
2021/11/11 10:54:31.291 [] [lettuce-epollEventLoop-8-3] [TRACE] i.l.c.p.CommandHandler - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800, epid=0x4b, chid=0x4b] Buffer: *1^M
2021/11/11 10:54:31.291 [] [lettuce-epollEventLoop-8-3] [DEBUG] i.l.c.p.CommandHandler - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800, epid=0x4b, chid=0x4b] Stack contains: 1 commands
2021/11/11 10:54:31.291 [] [lettuce-epollEventLoop-8-3] [DEBUG] i.l.c.p.CommandHandler - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800, epid=0x4b, chid=0x4b] Completing command LatencyMeteredCommand [type=XREAD, output=StreamReadOutput [output=[], error='null'], commandType=io.lettuce.core.cluster.ClusterCommand]
2021/11/11 10:57:30.791 [] [lettuce-epollEventLoop-8-5] [DEBUG] i.l.c.p.DefaultEndpoint - [channel=0x3af3bd45, /10.172.149.67:41402 -> 10.172.147.53/10.172.147.53:6800, epid=0x4b] write() writeAndFlush command ClusterCommand [command=SubscriptionCommand [type=XREAD, output=StreamReadOutput [output=[], error='null'], commandType=io.lettuce.core.protocol.TracedCommand], redirections=0, maxRedirections=5]
```

<details>
<summary>Stack trace</summary>

```java
// your stack trace here;
```

</details>

#### Input Code

<details>
<summary>Input Code</summary>

```java
// your code here;

   public Flux<StreamMessage<String, String>> blockReadOnStream(String streamName, String marker, Duration blockTime,
        int limit) {
        return Mono.fromFuture(connectionAsyncPool.acquire())
            .flatMapMany(connection -> {
                XReadArgs readArgs = new XReadArgs().count(limit);
                if (!blockTime.isZero()) {
                    readArgs.block(blockTime);
                }
                return connection
                    .reactive()
                    .xread(readArgs, XReadArgs.StreamOffset.from(streamName, marker))
                    .doFinally(ignore -> connectionAsyncPool.release(connection));
            });
    }
```

</details>

#### Environment
6.1.4.RELEASE
- Lettuce version(s): [e.g. 5.0.0.RELEASE, 4.2.2.Final]
- Redis version: [e.g. 4.0.9]

**Comment by mp911de** [comment]
Care to elaborate what your question is? One thing to note is that `Mono.fromFuture(connectionAsyncPool.acquire())` acquires the connection eagerly (when calling `blockReadOnStream`) while the actual Redis command is deferred until the subscription of the `Publisher`.

**Comment by xingzhang8023** [comment]
> Care to elaborate what your question is? One thing to note is that `Mono.fromFuture(connectionAsyncPool.acquire())` acquires the connection eagerly (when calling `blockReadOnStream`) while the actual Redis command is deferred until the subscription of the `Publisher`.

I want concurrent calls to xreads with blocks, and each client uses a new socket instead of reusing sockets.

**Comment by xingzhang8023** [comment]
> > Care to elaborate what your question is? One thing to note is that `Mono.fromFuture(connectionAsyncPool.acquire())` acquires the connection eagerly (when calling `blockReadOnStream`) while the actual Redis command is deferred until the subscription of the `Publisher`.
> 
> I want concurrent calls to xreads with blocks, and each client uses a new socket instead of reusing sockets.
I want every client to be able to block the server waiting for notifications.

**Comment by xingzhang8023** [comment]
The reason is that reactor-http-epoll invokes RedisPublisher cancel, but the xread command is still waiting to be returned in the channel stack.  In this case, if the channel is reused to send new command, new command will wait the xread response.
